benches/src/bin/collect.rs

```diff
@@ -570,9 +581,10 @@ impl State {
         let relative = groups.into_iter().filter_map(|(name, samples)| {
             let relative = samples
                 .into_iter()
-                .find(|sample| sample.ends_with(&name))
```

Comment:
This was causing the problem with new benchmarks not appearing in the final output. 

Benchmarks have a group name and a benchmark name, and together they form a sample. E.g., a `foo_group` group with benchmarks `test_n_1`, `test_n_10`, and `test_n_100`:

```rust
pub fn foo_tests(c: &mut Criterion) {
    let mut group = c.benchmark_group("foo_group");

    group.bench_function("test_n_1", |b| {
        b.iter(|| {...})
    });

    group.bench_function("test_n_10", |b| {
        b.iter(|| {...})
    });

    group.bench_function("test_n_100", |b| {
        b.iter(|| {...})
    });

    group.finish();
}

```

 would produce the following samples: 
- `foo_group/test_n_1`
- `foo_group/test_n_10` 
- `foo_group/test_n_100`

The above check was testing that the sample _ends_ with the group name. The above example benchmarks would fail this check and be excluded by the `find` because they instead _start_ with the group name.